### PR TITLE
[Post Ray 2.4 Release] Update Ray versions to Ray 2.4.0

### PR DESCRIPTION
**Pull request by jjyao**
## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: docs/guidance/aws-eks-iam.md [hunk 1]
```diff
@@ -24,16 +24,16 @@ s3.download_file(bucket_name, key, filename)
 ```
 
 The root cause is that the version of `boto3` in the Ray image is too old. To elaborate, `rayproject/ray:2.3.0` provides boto3 version 1.4.8 (Nov. 21, 2017),
-a more recent version (1.26) is currently available as per https://pypi.org/project/boto3/#history. The `boto3` 1.4.8 does not support to initialize the security credentials automatically in some cases (e.g. `AssumeRoleWithWebIdentity`). 
+a more recent version (1.26) is currently available as per https://pypi.org/project/boto3/#history. The `boto3` 1.4.8 does not support to initialize the security credentials automatically in some cases (e.g. `AssumeRoleWithWebIdentity`).
 
 ```shell
-# image: rayproject/ray:2.3.0
+# image: rayproject/ray:2.4.0
```
Comment (kevin85421) [inline docs/guidance/aws-eks-iam.md:30]: Hi @architkulkarni, is boto3 updated in `rayproject/ray:2.4.0`?

**Review by kevin85421 — APPROVED** [review]
LGTM. If boto3 is not upgraded in Ray 2.4.0, feel free to merge this PR.

File: docs/guidance/aws-eks-iam.md (hunk 1 above)
Comment (architkulkarni) [reply docs/guidance/aws-eks-iam.md:30]: It isn't.

**Comment by kevin85421** [comment]
The failure of "Compatibility Test - Nightly" is unrelated to this PR. The root cause is https://github.com/ray-project/ray/issues/34799. Merge this PR.

**Pull request merged**